plugin.ts

```diff
@@ -14,16 +16,18 @@ const DEFAULT_SETTINGS: WaybackMachinePluginSettings = {
 export class WaybackMachinePlugin extends Plugin {
   public settings: WaybackMachinePluginSettings = DEFAULT_SETTINGS;
   private isActive: boolean = false;
+  private linkReplacer!: LinkReplacer;
+  private status!: StatusReporter;
```

Comment:
I rather we not use the `!` operator because I cannot say for certain that it is always defined (although it appears so in the current context). We should leave it as a possibility that it can be undefined and leave it to callers to check.

Perhaps we should create a `setStatus` helper method which encapsulate the checking function. If unset, it can initialise it automatically. I think that is more maintainable.